app/script/conversation/ConversationRepository.coffee

```diff
@@ -1327,17 +1329,54 @@ class z.conversation.ConversationRepository
-    .then (response) =>
-      @_handle_client_mismatch conversation_id, response
-      return response
-    .catch (error) =>
-      throw error if not error.missing
+    return Promise.resolve()
+      .then =>
+        conversation_et = @find_conversation_by_id conversation_id
+        if conversation_et.verification_state() is z.conversation.ConversationVerificationState.DEGRADED
+          users_with_unverified_clients = conversation_et.get_users_with_unverified_clients()
+          user_ids_with_unverified_clients = users_with_unverified_clients.map (user_et) ->
+            return user_et.id
+          return @_grant_outgoing_message conversation_id, generic_message, user_ids_with_unverified_clients
+      .then =>
+        return @conversation_service.post_encrypted_message conversation_id, payload, precondition_option
+      .then (response) =>
+        @_handle_client_mismatch conversation_id, response
+        return response
+      .catch (error) =>
+        updated_payload = undefined
+
+        if error.missing
+          return @_handle_client_mismatch conversation_id, error, generic_message, payload
+          .then (payload_with_missing_clients) =>
+            updated_payload = payload_with_missing_clients
+            return @_grant_outgoing_message conversation_id, generic_message, Object.keys(error.missing)
+          .then =>
+            @logger.info "Sending updated encrypted '#{generic_message.content}' message to conversation '#{conversation_id}'", updated_payload
+            return @conversation_service.post_encrypted_message conversation_id, updated_payload, true
+        else
+          throw error
 
-      return @_handle_client_mismatch conversation_id, error, generic_message, payload
-      .then (updated_payload) =>
-        @logger.info "Sending updated encrypted '#{generic_message.content}' message to conversation '#{conversation_id}'", updated_payload
-        return @conversation_service.post_encrypted_message conversation_id, updated_payload, true
+  _grant_outgoing_message: (conversation_id, generic_message, user_ids) =>
+    conversation_et = @find_conversation_by_id conversation_id
+    return new Promise (resolve, reject) =>
+      if conversation_et.verification_state() is z.conversation.ConversationVerificationState.UNVERIFIED
+        resolve()
+      else if generic_message.content in ['cleared', 'confirmation', 'lastRead']
+        resolve()
+      else
+        send_anyway = false
+
+        @user_repository.get_users_by_id user_ids, (user_ets) ->
+          joined_usernames = z.util.LocalizerUtil.join_names user_ets
+
+          amplify.publish z.event.WebApp.WARNING.MODAL, z.ViewModel.ModalType.NEW_DEVICE,
+            data: joined_usernames
+            action: ->
+              send_anyway = true
+              conversation_et.verification_state z.conversation.ConversationVerificationState.UNVERIFIED
+              resolve()
+            close: ->
```

Comment:
Modals support a secondary action. Then you do not explicitly need to reject the error actually. If you do reject, you need the catch the error.